### PR TITLE
Lobbyist employer scraper

### LINKED ISSUE #29: Scrape lobbyist employer expenditures
The current lobbyist scrape captures employer name (ClientName), but there is some additional metadata we could capture: https://docs.google.com/spreadsheets/d/1-c2Ony5hGjpchOfwPKhYyoJ9FPji0dfkgt0mTcBS9mo/edit?usp=sharing

Clarify with Marjorie which employer metadata lobbyist scrapes should include, then capture it.

### PR DESCRIPTION
**Pull request by antidipyramid**
## Overview

This PR adds a lobbyist employer scraper to the existing lobbyist scraper in `scrapers/lobbyist/scrape_filings.py`. The after scraping both individual lobbyist filings and lobbyist employer filings, the pipeline is unchanged.

Depends on #30 

Closes #29 

## Testing

Run `make data/processed/lobbyist_expenditures.csv`

### CODE REVIEW

File: scrapers/lobbyist/scrape_filings.py [hunk 1]
```diff
@@ -1,79 +1,139 @@
...
+            "LobbyistID": id,
+            "LobbyistVersionID": version,
+        }
+
+    def scrape(self, id, version):
+        return super().scrape(self.__class__.url, id, version)
+
+
+@click.command()
+@click.option("--employer", "is_employer_scrape", is_flag=True)
```
Comment (antidipyramid) [inline scrapers/lobbyist/scrape_filings.py:79]: Used a flag to hook into the existing filing scraper.

File: lobbyists.mk [hunk 2]
```diff
@@ -60,12 +60,24 @@ data/intermediate/lobbyists.csv : data/raw/lobbyists.csv
 data/intermediate/clients.csv : data/raw/clients.csv
 	csvsql --query "SELECT ClientID, ClientVersionID, MAX(ClientName) AS ClientName FROM STDIN GROUP BY ClientID" < $< > $@
 
-data/intermediate/filings.csv : data/raw/lobbyists.csv
-	csvsql --query "SELECT DISTINCT MemberID, MemberVersionID FROM STDIN" < $< | \
+# Concatenate individual lobbyist and lobbyist employer filings
+data/intermediate/filings.csv : data/intermediate/employer_filings.csv data/intermediate/individual_filings.csv
+	tail -n +2 data/intermediate/individual_filings.csv > data/intermediate/individual_filings_rows.csv; \
+	cat data/intermediate/employer_filings.csv data/intermediate/individual_filings_rows.csv > $@
```
Comment (hancush) [inline lobbyists.mk:66]: How about [`csvstack`](https://csvkit.readthedocs.io/en/latest/scripts/csvstack.html) for this?

File: scrapers/lobbyist/scrape_filings.py [hunk 3]
```diff
@@ -1,79 +1,139 @@
...
+            "PageNumber": 1,
+            "PageSize": 1000,
+            "SortDir": "ASC",
+            "SortedBy": "",
+            "Year": "",
+            "ClientID": id,
+            "ClientVersionID": version,
+        }
+
+    def scrape(self, id, version):
```
Comment (hancush) [inline scrapers/lobbyist/scrape_filings.py:56]: Why is this call signature different from the abstract base class?

**Review by hancush — CHANGES_REQUESTED** [review]
Looking legit! One suggestion inline, and a question about how you organized your abstract base class and inheriting classes.

File: scrapers/lobbyist/scrape_filings.py (hunk 3 above)
Comment (antidipyramid) [reply scrapers/lobbyist/scrape_filings.py:56]: I couldn't figure out a clean way for the `LobbyistScraper` base class to reference the `url` attribute in a child class. 

[Turns out you can't stack @classmethod and @property to define a static class attribute as of 3.11.](https://docs.python.org/3/whatsnew/3.11.html#language-builtins)

If I'm overlooking something, please let me know.

File: lobbyists.mk (hunk 2 above)
Comment (antidipyramid) [reply lobbyists.mk:66]: Good idea!

File: scrapers/lobbyist/scrape_filings.py (hunk 3 above)
Comment (hancush) [reply scrapers/lobbyist/scrape_filings.py:56]: How about stacking @property and @abstractmethod? https://docs.python.org/3/library/abc.html#abc.abstractproperty

Comment (antidipyramid) [reply scrapers/lobbyist/scrape_filings.py]: Good idea!

File: scrapers/financial_disclosure/scrape_independent_expenditures.py [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+import logging
+import scrapelib
+import sys
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(stream=sys.stderr, level=logging.DEBUG)
+
+
+class IndependentExpenditureScraper(scrapelib.Scraper):
+    election_years = ("2021", "2022", "2023", "2024")
```
[the remark this answers is not in the document]
Comment (antidipyramid) [reply scrapers/financial_disclosure/scrape_independent_expenditures.py:12]: Good idea-- yes, they're also hard coded in `scrape_offices.py`. I'll open an issue.

**Pull request merged**